app/models/mortality_tracking.rb

```diff
@@ -43,4 +43,34 @@ class MortalityTracking < ApplicationRecord
     COMMENTS: "Comments"
   }
 
+  def self.create_from_csv_data(attrs)
+    attrs['shl_case_number']                     = attrs.delete('shl_number')
+    attrs['number_morts']                     = attrs.delete('_morts')
+
+    # Attempt to convert provided csv attributes
+    # to the proper format. Dates are provided as
+    # strings and must be converted to a datetime.
+    #
+    # Note - DateTime.strptime will raise a ArgumentError
+    # if there is a issue converting.
+    attrs['spawning_date'] = DateTime.strptime(attrs.delete('spawning_date'), '%m/%d/%y')
+    attrs['mortality_date'] = DateTime.strptime(attrs.delete('mortality_date'), '%m/%d/%y')
```

Comment:
https://github.com/wesleyeewong/abalone/blob/e3b30aa4374657e79263968a84170de2eaa498e0/app/models/untagged_animal_assessment.rb#L71-L80

If you add this to your code, the errors will bubble up to user level, where they can discern what's wrong with the csv.